File: telegram-bot-core/src/main/java/ru/tggc/telegrambotcore/registry/resolver/MessageIdResolver.java
package ru.tggc.telegrambotcore.registry.resolver;

import org.springframework.stereotype.Component;
import ru.tggc.telegrambotcore.annotation.params.MessageId;

import java.lang.reflect.Parameter;

@Component
public class MessageIdResolver implements ParameterResolver<Integer> {

    @Override
    public boolean supports(Parameter p) {
        return p.isAnnotationPresent(MessageId.class);
    }

    @Override
    public Integer resolve(Parameter parameter, HandlerCtx ctx) {
        return ctx.messageId();
    }
}

File: telegram-bot-core/src/main/kotlin/ru/tggc/telegrambotcore/registry/resolver/MessageIdResolver.kt
package ru.tggc.telegrambotcore.registry.resolver

import org.springframework.stereotype.Component
import ru.tggc.telegrambotcore.annotation.params.MessageId
import java.lang.reflect.Parameter

@Component
class MessageIdResolver : ParameterResolver<Int?> {
    override fun supports(p: Parameter?): Boolean? = p?.isAnnotationPresent(MessageId::class.java)

    override fun resolve(parameter: Parameter?, ctx: HandlerCtx): Int? = ctx.messageId
}
